### ISSUE TITLE
Improvement: ass VSTS as a CI Platform

### ISSUE THREAD

**Issue by andylockran** [outofcoffee/corebot#50]

**Comment by andylockran** [comment]
Oh how I wish it was possible to edit the title on Mobile :)

**Comment by outofcoffee** [comment]
FTFY @andylockran ;-)

Any links to APIs you can provide? Hopefully shouldn't be too large a job assuming it takes a similar approach to other CI platforms.

**Comment by outofcoffee** [comment]
Looks like might be 'Queue a build' on https://www.visualstudio.com/en-us/docs/integrate/api/build/builds

```
POST https://{instance}/DefaultCollection/{project}/_apis/build/builds?api-version={version}
```

Need to figure out:
1. how we test this
2. authentication

**Comment by outofcoffee** [comment]
Regarding auth, it looks like we could use HTTP Basic over HTTPS, with a personal access token: https://docs.microsoft.com/en-us/vsts/integrate/get-started/authentication/pats

@andylockran would this work for your use case?

**Comment by andylockran** [comment]
Yes, I’m building out and working out the endpoints. I will submit a PR, not just make you work on the weekends ;)

**Comment by andylockran** [comment]
Aww Pete I just found evidence of our earlier collab
